### PR TITLE
Fix/tbd 6336

### PR DESCRIPTION
**Pull request by mtuhai**
**Please check if the PR fulfills these requirements**

- [X] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format) ?
- [X] Unit tests for the Java changes have been added (for bug fixes / features) ?
- [X] TUJ for the JavaJet changes have been added (for bug fixes / features) ?
- [X] Docs have been added / updated (for bug fixes / features) ?
- [X] The new code does not introduce new technical issues

**What is the current behavior?** (You can also link to an open issue here)
https://jira.talendforge.org/browse/TBD-6336

**What is the new behavior?**

### CODE REVIEW

File: main/plugins/org.talend.designer.components.bigdata/components/tPigStoreResult/tPigStoreResult_main.javajet [hunk 1]
```diff
@@ -93,7 +97,7 @@
 	boolean generateRegisterJarCodeForHCatalog = ("HCatStorer".equals(function) && !defineJarsToRegister);
 	boolean generateRegisterJarCode = registerJar.size() > 0;
 	
-	String hcatPackage = (pigDistrib.pigVersionPriorTo_0_12() && !isCustom) ? "org.apache.hcatalog.pig" : "org.apache.hive.hcatalog.pig";
+	String hcatPackage = (!isCustom && pigDistrib.pigVersionPriorTo_0_12()) ? "org.apache.hcatalog.pig" : "org.apache.hive.hcatalog.pig";
```
Comment (rdubois) [inline main/plugins/org.talend.designer.components.bigdata/components/tPigStoreResult/tPigStoreResult_main.javajet]: `pigDistrib` can be null. Can you please secure this part of code?

Comment (lbourgeois) [reply main/plugins/org.talend.designer.components.bigdata/components/tPigStoreResult/tPigStoreResult_main.javajet]: if `pigDistrib==null` `isCustom` will be true so `pigDistrib.pigVersionPriorTo_0_12()` won't be evaluated

Comment (rdubois) [reply main/plugins/org.talend.designer.components.bigdata/components/tPigStoreResult/tPigStoreResult_main.javajet]: +1, my bad.

File: main/plugins/org.talend.designer.components.bigdata/components/tPigStoreResult/tPigStoreResult_main.javajet [hunk 2]
```diff
@@ -54,23 +54,27 @@
...
+		if (initDistro && (distribution != null) && !"".equals(distribution)) {
+		    pigDistrib = (org.talend.hadoop.distribution.component.PigComponent) org.talend.hadoop.distribution.DistributionFactory.buildDistribution(distribution, pigVersion);
+		}
 	} catch (java.lang.Exception e) {
 		e.printStackTrace();
 		return "";
 	}
 
-	boolean isCustom = pigDistrib instanceof org.talend.hadoop.distribution.custom.CustomDistribution;
+	boolean isCustom = (pigDistrib == null) || (pigDistrib instanceof org.talend.hadoop.distribution.custom.CustomDistribution);
```
Comment (rdubois) [inline main/plugins/org.talend.designer.components.bigdata/components/tPigStoreResult/tPigStoreResult_main.javajet]: If `pigDistrib` is null, it doesn't necessarily mean that we are in custom mode.

**Comment by rdubois** [comment]
By looking at the code, the variable `pigDistrib` and `isCustom` are only used to create the variable `hcatPackage`, and this variable is only used once in the code, when the HCatStorer is selected, so we could simply move all of this code:

```
	String distribution = ElementParameterParser.getValue(node, "__DISTRIBUTION__");
	String pigVersion = ElementParameterParser.getValue(node, "__HBASE_VERSION__");
	org.talend.hadoop.distribution.component.PigComponent pigDistrib = null;
	try {
		pigDistrib = (org.talend.hadoop.distribution.component.PigComponent) org.talend.hadoop.distribution.DistributionFactory.buildDistribution(distribution, pigVersion);
	} catch (java.lang.Exception e) {
		e.printStackTrace();
		return "";
	}

	boolean isCustom = pigDistrib instanceof org.talend.hadoop.distribution.custom.CustomDistribution;

	String hcatPackage = (pigDistrib.pigVersionPriorTo_0_12() && !isCustom) ? "org.apache.hcatalog.pig" : "org.apache.hive.hcatalog.pig";

```

at line 346, in the `if("HCatStorer".equals(function)) {` condition. What do you think ?

**Pull request closed without merging**